client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,24 +1,220 @@
+	onAnnualPlanClick: () => void;
+}
+
+const UpgradePlanPeriodSwitcher = ( props: SwitcherProps ) => {
+	const { __ } = useI18n();
+
+	if ( config.isEnabled( 'migration-flow/introductory-offer' ) ) {
+		return null;
+	}
+
+	const { selectedPlan, onMonthlyPlanClick, onAnnualPlanClick } = props;
+
+	return (
+		<div className="import__upgrade-plan-period-switcher">
+			<ButtonGroup>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS_MONTHLY } ) }
+					onClick={ onMonthlyPlanClick }
+				>
+					{ __( 'Pay monthly' ) }
+				</Button>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS } ) }
+					onClick={ onAnnualPlanClick }
+				>
+					{ __( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+const Badge = styled.div< { isForIntroOffer?: boolean; isHidden?: boolean } >`
+	text-align: center;
+	white-space: nowrap;
+	font-size: 0.75rem;
+	line-height: 1.25rem;
+	border-radius: 4px;
+	height: 21px;
+	display: inline-block;
+	width: fit-content;
+	letter-spacing: 0.2px;
+	font-weight: 500;
+	text-align: center;
+	padding: 0 12px;
+	background-color: #e6d1ff;
+	color: #350e63;
+	text-transform: none;
+`;
+
+const PricesGroup = styled.div`
+	justify-content: flex-end;
+	display: flex;
+	flex-direction: row-reverse;
+	align-items: flex-end;
+	gap: 4px;
+`;
+
+interface PriceWithIntroductoryOfferProps {
+	originalMonthlyPrice: number;
+	introOfferMonthlyPrice: number;
+	currencyCode?: string;
+}
+
+const PriceWithIntroductoryOffer = ( props: PriceWithIntroductoryOfferProps ) => {
+	const { originalMonthlyPrice, introOfferMonthlyPrice, currencyCode } = props;
+	const translate = useTranslate();
+
+	return (
+		<>
+			<Badge>{ translate( 'One time offer' ) }</Badge>
+			<PricesGroup>
+				<PlanPrice
+					rawPrice={ originalMonthlyPrice }
+					currencyCode={ currencyCode }
+					original
+					isSmallestUnit
+				/>
+				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
+			</PricesGroup>
+		</>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	formattedOriginalPrice?: string;
+	formattedIntroOfferPrice?: string;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		formattedOriginalPrice,
+		formattedIntroOfferPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled && ! isIntroOfferComplete && introOfferMonthlyPrice && originalMonthlyPrice;
+
+	const price = showIntroOffer ? (
+		<PriceWithIntroductoryOffer
+			originalMonthlyPrice={ originalMonthlyPrice }
+			introOfferMonthlyPrice={ introOfferMonthlyPrice }
+			currencyCode={ currencyCode }
+		/>
```

Comment:
Minor non-blocking note: I am not sure that shifting this into a new component is a good move, especially from a cognitive flow perspective, as the interior component is just returning a fragment. We could inline that and do away with the wrapper component and props interface.
```suggestion
		<>
			<Badge>{ translate( 'One time offer' ) }</Badge>
			<PricesGroup>
				<PlanPrice
					rawPrice={ originalMonthlyPrice }
					currencyCode={ currencyCode }
					original
					isSmallestUnit
				/>
				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
			</PricesGroup>
		</>
```